R/SpectraWrapper.R

```diff
@@ -1,7 +1,168 @@
+  genesets_json = paste0(seuratToAdataDir, "/genesets.json")
+  variable_features_file = paste0(seuratToAdataDir, "varfeats.csv")
+
+  ProduceGenesetJSONfile(inputJSONfile, immportfile, genesets_json)
+
+  #copy run_Spectra.py in inst/scripts and supply custom arguments
+  str <- readr::read_file(system.file("scripts/run_Spectra.py", package = "CellMembrane"))
+  script <- tempfile()
+  readr::write_file(str, script)
+
+  # run_Spectra(features_file, metadata_file, gex_datafile, genesets_json)
+  newstr <- paste0("run_Spectra(features_file = '", variable_features_file,
+                   "', gex_datafile = '", gex_datafile,
+                   "', metadata_file = '", metadata_file,
+                   "', genesets_json = '", genesets_json,
+                   "', seuratToAdataDir = '", seuratToAdataDir,
+                   "')")
+
+  #write the new arguments into the script and execute
+  readr::write_file(newstr, script, append = TRUE)
+  system2(reticulate::py_exe(), script)
+  seuratObj <- SpectraToSeuratReduction(seuratObj, seuratToAdataDir, assayName, reduction.name = 'spectra', reduction.key = 'Spectra_')
+  seuratObj <- SpectraToSeuratMetadata(seuratObj, paste0(seuratToAdataDir, "/SPECTRA_cell_scores.csv"))
+  return(seuratObj)
+}
+
+
+#' @title SeuratToAData
+#' @param seuratObj The Seurat object to be written into a 10x file format
+#' @param seuratToAdataDir The local path to write the output files.
+#' @param assayName The name of the gene expression data assay.
+#' @description A wrapper function to prepare a Seurat object for Conga.
+#' @export
+SeuratToAData <- function(seuratObj,
+                          seuratToAdataDir,
+                          assayName = 'RNA') {
+  if (!dir.exists(seuratToAdataDir)) {
+    dir.create(seuratToAdataDir, recursive = T)
+  }
+  seuratObj <- Seurat::FindVariableFeatures(seuratObj)
+  write.table(Seurat::VariableFeatures(seuratObj),
+              R.utils::getAbsolutePath(paste0(seuratToAdataDir, "/varfeats.csv"),
+                                       mustWork = FALSE), row.names = FALSE, col.names = FALSE)
+
+  DropletUtils::write10xCounts(x = Seurat::GetAssayData(seuratObj, assay = assayName, slot = 'data'),
+                               path = R.utils::getAbsolutePath(paste0(seuratToAdataDir, "/GEX.h5"), mustWork = FALSE),
+                               overwrite = TRUE)
+  data.table::fwrite(seuratObj@meta.data,
+                     R.utils::getAbsolutePath(paste0(seuratToAdataDir, "/metadata.tsv"),
+                                              mustWork = FALSE), row.names = TRUE, col.names = TRUE)
+
+}
+
+
+ProduceGenesetJSONfile <- function(inputpath, inputpath2, outputpath){
+  # read in gene sets
+  jsonin <- jsonlite::read_json(inputpath, simplifyVector = TRUE)
+  immport <- readRDS(inputpath2)
+  genesets <- RIRA::ListGeneSets()
+
+  geneset_auto <- list()
+  geneset_auto$global <- jsonin$global
+  for (i in c(1:length(genesets))){
```

Comment:
This loop needs some TLC - the gene sets in RIRA are probably parsable into their gross celltypes - if we have a constructor for this then we can enforce some basic structure here.